src/network/url_downloader.cpp

```diff
@@ -128,8 +129,11 @@ void mp::URLDownloader::download_to(const QUrl& url, const QString& file_name, i
 
 QByteArray mp::URLDownloader::download(const QUrl& url)
 {
-    return ::download(manager, timeout, url, [](QNetworkReply*, qint64, qint64) {}, [](QNetworkReply*, QTimer&) {},
-                      [] {});
+    // This will connect to the QNetworkReply::readReady signal and when emitted,
+    // reset the timer.
+    auto on_download = [](QNetworkReply* reply, QTimer& download_timeout) { download_timeout.start(); };
```

Comment:
Why are lines 111-114 needed there but not here?

Reply:
@ricab, Because in lines 111-114, we want to stop the timer to allow time reading from the `QNetworkReply` and write to the file.  Here, we are not doing that and calling `start()` just resets the timer.

Reply:
@townsend2010 OK thanks, I missed that and thought could be part of the resetting